File: src/main/java/com/sem/keeper/interceptor/UserDataAdder.java
package com.sem.keeper.interceptor;

import com.sem.keeper.repo.LoanRequestRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import org.springframework.stereotype.Service;
import org.springframework.ui.Model;
import org.springframework.web.servlet.HandlerInterceptor;
import org.springframework.web.servlet.ModelAndView;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

@Component
public class UserDataAdder implements HandlerInterceptor {

    @Autowired
    LoanRequestRepository loanRequestRepository;

    @Override
    public void postHandle(HttpServletRequest request, HttpServletResponse response, Object handler, ModelAndView modelAndView) throws Exception {
        if(modelAndView==null) return;
        modelAndView.addObject("loanReqCnt",4);
    }
}

File: src/main/java/com/sem/keeper/interceptor/UserDataAdder.kt
package com.sem.keeper.interceptor

import com.sem.keeper.repo.LoanRequestRepository
import org.springframework.stereotype.Component
import org.springframework.web.servlet.HandlerInterceptor
import org.springframework.web.servlet.ModelAndView
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

@Component
class UserDataAdder(val loanRequestRepository: LoanRequestRepository) : HandlerInterceptor {
    override fun postHandle(
        request: HttpServletRequest,
        response: HttpServletResponse,
        handler: Any,
        modelAndView: ModelAndView
    ) {
        modelAndView.addObject("loanReqCnt", 4)
    }
}
